spec/src/main/asciidoc/app-programming-model.adoc

```diff
@@ -510,6 +508,69 @@ public void init(@Metric(name="instances") Counter instances) {
 }
 ----
 
+=== Registering metrics dynamically
+In addition to declaring metrics via annotations, it is possible to dynamically (un)register metrics by 
+calling methods of a `MetricRegistry` object. While using annotations is generally the preferred approach,
+registering metrics dynamically can be useful in cases when the final list of metrics is not known in 
+advance (when the application is being coded) or when there are too many metrics similar in a way
```

Comment:
this sentence gets lost a bit starting with "similar in a way..."

Reply:
Reworded that a bit.